### DISCUSSION TITLE
User text chat messages

### DISCUSSION THREAD

**Discussion by benjamincloquet** [saul/demofile#203] [Questions] [answered]
Hello! First of all thanks for the project @saul, pretty much everything's working fine.
Small bug though, I can't seem to catch any events for user text messages in my matchmaking demos. I tried reusing your code :

     demoFile.userMessages.on("SayText", e => {
        console.log(formatSayText(0, e.text));
    });

    demoFile.userMessages.on("SayText2", e => {
        const nonEmptyParams = e.params.filter(s => s.length);
        const msgText = standardMessages[e.msgName];
        const formatted = msgText ?
        util.format.apply(null, [msgText].concat(nonEmptyParams)) :
        `${e.msgName} ${nonEmptyParams.join(" ")}`;
            console.log(formatSayText(e.entIdx, formatted));
    });

Nothing is caught at all. Any tips ? Thanks!

**Comment by master117** [comment]
Did you call listen first? https://saul.github.io/demofile/classes/_usermessages_.usermessages.html#listen

**Comment by saul** [comment]
Hmm listen should be called automatically. Do you have an example of a demo where your code isn’t working? Does your demo work with the dumpfile example?

**Comment by benjamincloquet** [comment]
@master117 : I'm not sure I understand where I'd call this function, if this can give you some insight, the following code works :
```
    demoFile.userMessages.on("WarmupHasEnded", event => {
        console.log("End of warmup");
    });
```

@saul : I've been doing tests on several demos where other events seem to work fine (see my reply to master117)
There's plenty of chat text (one player says "gg" at the start of round 1, for reference) in this demo : https://we.tl/t-um50GXJAa6

**Comment by saul** [comment, marked as answer]
I'm going to solve this in https://github.com/saul/demofile/issues/322. Apologies as I didn't realise that some messages are encrypted in matchmaking demos when you first raised the issue.

**Reply by benjamincloquet** [reply]
Dope! Glad you're still working on the project after all this time. Cheers!
